packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java

```diff
@@ -60,12 +67,39 @@ public ResponseEntity<DocumentAsset> getDocument(
 		@PathVariable("id") String id
 	) {
 		DocumentAsset document = proxy.getAsset(id).getBody();
-
-		// Add the S3 bucket url to each asset metadata
 		document.getAssets().forEach(asset -> {
-			asset.getMetadata().put("url", proxy.getDownloadUrl(id, asset.getFileName()).getBody().getUrl());
+			String url = proxy.getDownloadUrl(id, asset.getFileName()).getBody().getUrl();
+
+			// Add the S3 bucket url to each asset metadata
+			asset.getMetadata().put("url", url);
+
+			// if the asset os of type equation
+			if (asset.getAssetType().equals("equation") && asset.getMetadata().get("equation") == null) {
+				byte[] imagesByte = new byte[0];
+				try {
+					// Fetch the image from the URL
+					imagesByte = IOUtils.toByteArray(new URL(url));
+					// Encode the image in Base 64
+					String imageB64 = Base64.getEncoder().encodeToString(imagesByte);
+
+					// Send it to SKEMA to get the Presentation MathML equation
+					String equation = skemaUnifiedProxy.postImageToEquations(imageB64).getBody();
+
+					log.warn("Equation: {}", equation);
+
+					// Add the equations into the metadata
+					asset.getMetadata().put("equation", equation);
+
+				} catch (IOException e) {
+					e.printStackTrace();
```

Comment:
this should be done through the logging system
`log.error("Unable to blah blah blah", e);`